lib/spider/scripts/in_context_learning.py

```diff
@@ -57,15 +65,35 @@ def main(cfg) -> None:
         }
         for data_point in dev_json
     ]
+    if cfg.prompts.template.count("{}") > 2:
+        train_json = json.load(open(DATASET_PATH / "train_spider.json"))
+        train_json = [
+            {
+                "db_id": data_point["db_id"],
+                "question": data_point["question"],
+                "query": data_point["query"],
+            }
+            for data_point in train_json
+        ]
+        train_questions = [data_point["question"] for data_point in train_json]
+        top_k = int((cfg.prompts.template.count("{}") - 2) / 2)
```

Comment:
to make it a bit more robust to different prompts top_k can be added as a parameter in the config